internal/callbacker/dispatcher.go

```diff
@@ -0,0 +1,108 @@
+package callbacker
+
+import (
+	"sync"
+	"time"
+)
+
+type CallbackDispatcher struct {
+	c        CallbackerI
+	managers map[string]*sendManager
```

Comment:
As an alternative strategy, here you could use a `sync.Map`
Here is an example where it is used: https://github.com/bitcoin-sv/arc/blob/6669b2339f9b019dec239acc264bf0214d705e33/internal/metamorph/response_processor.go#L44

Then you don't need the mutex and can remove all the lock/unlock statements
```
	d.mu.Lock()
	defer d.mu.Unlock()
```

Reply:
Unfortunately, with sync.Map, I am unable to check if a manager exists and create one only if it doesn't, in an atomic way.

The code I was able to write is:
```go
func (d *CallbackDispatcher) GracefulStop() {
	d.managers.Range(func(key, value any) bool {
		m := value.(*sendManager)
		m.GracefulStop()

		return true
	})
}

func (d *CallbackDispatcher) dispatch(url, token string, dto *Callback) {
	m, loaded := d.managers.LoadOrStore(url, &sendManager{
		url:   url,
		c:     d.c,
		sleep: d.sleep,

		ch: make(chan *callbackEntry),
	})

	manager := m.(*sendManager)
	if !loaded {
		manager.run()
	}

	manager.Add(token, dto)
}
```

I don't think this code is simpler than what already exists, and  TBH I'm not a fan of the idea of creating new manager structure for every call.

Reply:
Ok. Understood 👍